reconcile/utils/terraform_client.py

```diff
@@ -575,34 +577,74 @@ def cleanup(self):
         for _, wd in self.working_dirs.items():
             shutil.rmtree(wd)
 
-    def _is_ignored_rds_modification(
+    def _get_pending_rds_modification(
         self, account_name: str, resource_name: str, resource_change: Mapping[str, Any]
-    ) -> bool:
+    ) -> list[str]:
         """
-        Determine whether the RDS resource changes are cases where a terraform apply
+        Determine whether the RDS resource changes are cases where a Terraform apply
```

Comment:
Since this method has changed significantly, it might be worth updating the docstring? It seems that the general pattern now is that a falsy value will be returned when apply can't be skipped, but a truthy value, including the names of the affected fields, will be returned when the apply can be skipped?

nit: I think the intent of all of this could possibly be made clearer by calling this `_rds_modifications_can_be_skipped`. The only thing you really lose is the list of `PendingModifiedValues`, but it's a debug log anyways. We could easily put a debug log in this function to get the same data while simplifying the other log message? Then this method just returns a `bool`.

Reply:
@steveteahan, no worries. It does make sense to make it only return an boolean and keep all the extra log messages internally. Bear with me while I update everything.

Reply:
Done. Thank you!